app/controllers/AddConfidentialInformationController.scala

```diff
@@ -22,56 +22,31 @@ import controllers.actions._
 import forms.AddConfidentialInformationFormProvider
 import javax.inject.Inject
 import models.Mode
+import models.requests.DataRequest
 import navigation.Navigator
-import pages._
+import pages.ProvideGoodsNamePage
 import play.api.data.Form
-import play.api.i18n.I18nSupport
 import play.api.mvc.MessagesControllerComponents
-import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import play.twirl.api.HtmlFormat
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
+import navigation.Journey
 
 class AddConfidentialInformationController @Inject()(appConfig: FrontendAppConfig,
-                                                     dataCacheConnector: DataCacheConnector,
-                                                     navigator: Navigator,
-                                                     identify: IdentifierAction,
-                                                     getData: DataRetrievalAction,
-                                                     requireData: DataRequiredAction,
-                                                     formProvider: AddConfidentialInformationFormProvider,
-                                                     val add_confidential_information: views.html.addConfidentialInformation,
-                                                     cc: MessagesControllerComponents
-                                                    )(implicit ec: ExecutionContext) extends FrontendController(cc) with I18nSupport {
-
-  val form: Form[Boolean] = formProvider()
-
-  def onPageLoad(mode: Mode) = (identify andThen getData andThen requireData) {
-    implicit request =>
-
-      val goodsName = request.userAnswers.get(ProvideGoodsNamePage).getOrElse("goods")
-      val preparedForm = request.userAnswers.get(AddConfidentialInformationPage) match {
-        case None => form
-        case Some(value) => form.fill(value)
-      }
-
-      Ok(add_confidential_information(appConfig, preparedForm, goodsName, mode))
-  }
-
-  def onSubmit(mode: Mode) = (identify andThen getData andThen requireData).async {
-    implicit request =>
-
-      val goodsName = request.userAnswers.get(ProvideGoodsNamePage).getOrElse("goods")
-      form.bindFromRequest().fold(
-        (formWithErrors: Form[_]) =>
-          Future.successful(BadRequest(add_confidential_information(appConfig, formWithErrors, goodsName, mode))),
-
-        (value) => {
-          val updatedAnswers = request.userAnswers.set(AddConfidentialInformationPage, value)
-          val answers = if (value) updatedAnswers else updatedAnswers.remove(ProvideConfidentialInformationPage)
-          dataCacheConnector.save(answers.cacheMap).map(
-            _ =>
-              Redirect(navigator.nextPage(AddConfidentialInformationPage, mode)(updatedAnswers))
-          )
-        }
-      )
+  val dataCacheConnector: DataCacheConnector,
```

Comment:
I think this was a regression in our merge - can we take out the vals again? @warrenskiwork1979

Reply:
ha ha I don't think all of our conflict fixes took Adam when looking through the messages file previously, I'll sort this out.